scrapers/AdultPrime.yml

```diff
@@ -1,31 +1,84 @@
+# The search by name does not return images by default. To get images you will need to uncomment the driver section at the end of this file.
+# Each click you uncomment will load another 4 images, however the search will be much slower, so it's disabled by default.
 name: "AdultPrime"
+sceneByName:
+  action: scrapeXPath
+  queryURL: "https://adultprime.com/studios/search?q={}"
+  scraper: sceneSearch
+
+sceneByQueryFragment:
+  action: scrapeXPath
+  queryURL: "{url}"
+  scraper: sceneScraper
+
 sceneByURL:
   - action: scrapeXPath
     url:
       - adultprime.com/studios/video/
     scraper: sceneScraper
+
 xPathScrapers:
+  sceneSearch:
+    common:
+      $scenes: //ul[@id='studio-videos-container']
+    scene:
+      Title:
+        selector: $scenes//span[contains(@class, 'description-title')]/text()
+      Date:
+        selector: $scenes//span[@class='description-releasedate']/text()
+        postProcess:
+          - parseDate: Jan 02, 2006
+      URL:
+        selector: $scenes//a[@class='absolute']/@href
+        postProcess:
+          - replace:
+            - regex: "^.signup.galleryId."
+              with: "https://adultprime.com/studios/video/"
+      Image:
+        selector: $scenes//div[contains(@class, 'ap-slider-img')]/@style
+        postProcess:
+          - replace:
+              - regex: .+url\("(.+)"\).+
+                with: $1
+
   sceneScraper:
     scene:
       Title: //h2[contains(@class,"update-info-title")]/text()
       Date:
-        selector: //p[@class="update-info-line regular"][i[contains(@class,"fa-calendar")]]/b[1]/text()
+        selector: //p[@class="update-info-line regular"][1]/b[1]/text()

```

Comment:
were there cases that the original was not working ?

Reply:
Was a long time ago, so not sure, but I'm guessing there was a scene with no date, and so it tried to load whatever the next field was as the date.